src/fees/saga.ts

```diff
@@ -29,12 +29,17 @@ const TAG = 'fees/saga'
 export const SWAP_FEE_ESTIMATE_MULTIPLIER = 8
 export const SWAP_CELO_FEE_ESTIMATE_MULTIPLIER = 3 * SWAP_FEE_ESTIMATE_MULTIPLIER
 
-export interface FeeInfo {
+export type FeeInfo = {
   fee: BigNumber
   gas: BigNumber
   gasPrice: BigNumber
-  feeCurrency: string | undefined
-}
+} & (
```

Comment:
I know I probly suggested this discriminated union trick, but it seems like it forced you to use a bunch of clunky ternaries, which will slow things down later if we want to make feeCurrency optional even when feeTokenId is defined, for instance, to support native assets that don't have an address. (since you have things like `if (feeCurrency && feeTokenId) { ... }` in a few places just to avoid type errors, and each of those would need to be redone)

Reply:
might be better to just use `feeCurrency?: string; feeTokenId?: string`